tests/trainer/test_trainer.py

```diff
@@ -1969,3 +1970,52 @@ def training_step(self, batch, batch_idx):
     # simulate random failure in training_step on rank 0
     with pytest.raises(DeadlockDetectedException, match="CustomException"):
         trainer.fit(model)
+
+
+@RunIf(min_gpus=1)
+def test_multiple_trainer_constant_memory_allocated(tmpdir):
+    """
+    This tests ensures calling the trainer several times reset the memory back to 0.
+    """
+
+    class TestModel(BoringModel):
+
+        def __init__(self):
+            super().__init__()
+            self._example_input_array = torch.zeros((2, 32))
+
+        @property
+        def example_input_array(self):
+            return self._example_input_array
```

Comment:
You don't need this if you set it directly in __init__ without the underscore

```suggestion
```

Reply:
Removed !